### ISSUE TITLE
Java 9 String getBytes performance investigation

### ISSUE THREAD

**Issue by andrewcraik** [eclipse/openj9#375]
The getBytes implementation in Java 9 may need to allocate a new byte[] array matching the string length when the backing array is longer than the string length. There is an idea of storing the shortened byte array back to the string. While correct this could increase memory footprint, but could also increase throughput of subsequent getBytes calls. This item is to track doing some kind of performance investigation to determine whether updating the String's value field to hold the shortened string increases performance and/or adversely affects memory consumption.

**Comment by DanHeidinga** [comment]
@andrewcraik @fjeremic Is this still relevant?

**Comment by andrewcraik** [comment]
Yes - we haven't continued to do perf investigation on this so there may still be a perf opportunity to look at here. Focus has been on other areas.
